File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> b[1] - a[1]);
        for (int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        for (int key : map.keySet()) {
            pq.add(new int[]{key, map.get(key)});
        }
        int[] res = new int[k];
        while (--k >= 0) {
            res[k] = pq.poll()[0];
        }
        return res;
    }
}

File: Data Structures & Algorithms/merge-strings-alternately/submission-0.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i = j = 0
        res = str()
        while i < len(word1) and j < len(word2):
            res += word1[i]
            res += word2[j]
            i += 1
            j += 1
        
        if i < len(word1):
            res += word1[i:len(word1)]
        elif j < len(word2):
            res += word2[j:len(word2)]

        return res    
        